### ISSUE TITLE
Accept application/zip files from front-end

### ISSUE THREAD

**Issue by jdegoes** [quasar-analytics/quasar#1618]
The front-end cannot know the type of all files inside a ZIP archive, and, indeed, there may be many types, rather than a single type. Further, by HTTP spec, the content type of the request must necessarily be the content type of the content, not the content of files inside the decoded content.

For these reasons, the backend will need to encode and decode whatever metadata it requires into and out of the ZIP archive that it generates.

If we switch to using EJSON, this may have a straightforward solution. Otherwise some hack may be required, such as creating a hidden `.quasar-metadata.json` file which contains metadata annotations for all flat children in a directory.

**Comment by mossprescott** [comment]
It seems like the zip format has some support for metadata, but it’s not very well specified, so I think it's probably safer to add the suggested magic dot-file with the information we need.

So here's the plan:

- [ ] make sure `Content-Type` is `application/zip` in both directions.
- [ ] add `.quasar-metadata.json` at the root of the zip archive, containing whatever additional info is needed to write the files.
- [ ] parse, validate, and use the metadata when uploading
- [ ] add some tests for alternative formats
- [ ] test upload with heterogenous formats (hand-assembled zip)

The metadata file will look something like this (not that anyone apart from quasar will need to understand it):
```json
{
  "foo": {
    "Content-Type": "application/ldjson; mode=\"precise\""
  },
  "bar/baz": {
    "Content-Type": "application/ldjson; mode=\"precise\""
  }
}
```

**Comment by jdegoes** [comment]
This could provide a path to saving / restoring mounts, too.

**Comment by jedesah** [comment]
Fixed by #1883
